QuickBrick/plugins/TViOSLogin/src/Common/Utils/index.js

```diff
@@ -27,3 +30,48 @@ export async function setToLocalStorage(key, value) {
 export async function getFromLocalStorage(key) {
   return storage.getItem(key);
 }
+
+function validateStyles(pluginData) {
+  const keys = Object.keys(pluginData);
+  keys.forEach((key) => {
+    const type = key.split('_').pop();
+    if (type === 'fontsize' || type === 'fontcolor') {
+      validateKey(type, key, pluginData);
+    }
+  });
+}
+
+function validateKey(type, key, pluginData) {
+  const keysValidation = {
+    fontsize: validateFontsize,
+    fontcolor: validateFontcolor
+  };
+
+  return keysValidation[type](key, pluginData);
+}
+
+const validateFontsize = (key, pluginData) => {
+  const value = pluginData[key];
+  const keyname = R.replace('_fontsize', '', key);
+
+  const num = Number(value);
+  pluginData[key] = Number.isFinite(num) ? num : fontsize[keyname];
+};
+
+const validateFontcolor = (key, pluginData) => {
+  const value = pluginData[key];
+  const keyname = R.replace('_fontcolor', '', key);
+
+  pluginData[key] = (value !== undefined && value !== null) ? value : fontcolor[keyname];
+};
+
+export const isHomeScreen = (navigator, homeScreen) => {
+  const homeScreenPath = `/river/${homeScreen.id}`;
+  return navigator.currentRoute === homeScreenPath;
```

Comment:
if (for any reason) navigator is undefined, javascript crash

Reply:
If navigator is undefined, the whole app will crash, because it's QuickBrick Core functionality and all screens transactions are based on it